src/ape_http/providers.py

```diff
@@ -1,15 +1,23 @@
-from typing import Iterator
```

Comment:
Main changes of PR are in this file.

We just stop using the util method and use a more custom error handling approach to geth, defined in the util method at the bottom of the module.